### PR TITLE
build(deps, deps-dev): upgrade major versions for dependencies of `@superset/embedded-sdk`

### PR DESCRIPTION
**Pull request by hainenber**
build(deps, deps-dev): upgrade major versions for dependencies of `@superset/embedded-sdk`

### SUMMARY

- Upgrade major versions for dependencies of `@superset/embedded-sdk`
- Fix `npm audit` findings

### ADDITIONAL INFORMATION

- [x] Has associated issue: Fixes https://github.com/apache/superset/issues/30241
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by rusackas** [comment]
Thanks @hainenber - if you bump the patch version in package.json, this should release the new version automatically (or so it seems). This also seems to need a rebase, since I already bumped it in another PR.

File: superset-embedded-sdk/package.json [hunk 1]
```diff
@@ -34,20 +34,20 @@
   ],
   "dependencies": {
     "@superset-ui/switchboard": "^0.20.2",
-    "jwt-decode": "^3.1.2"
+    "jwt-decode": "^4.0.0"
```
[the remark this answers is not in the document]
Comment (rusackas) [reply superset-embedded-sdk/package.json:37]: Agreed. Thanks for looking into it.

**Review by rusackas — APPROVED** [review]
Fantastic, thanks! Once this is rebased, I think it'll be good to go.

**Comment by rusackas** [comment]
We're going to have to make you a committer, so you can make branches on `superset` and review PRs, and I can repay the favor (at least a little) by helping to rebase things for you when needed ;)

**Comment by hainenber** [comment]
Rebased :D

**Comment by hainenber** [comment]
Thanks @rusackas! Great power comes with great responsibility for me from now on :D

**Pull request merged**